netam/oe_plot.py

```diff
@@ -72,8 +89,40 @@ def annotate_sites_df(
 
         is_cdr_col.append(pcp_sites_cdr_annotation(pcp_row))
 
+        if add_codons_aas:
+            parent_codons = []
+            parent_aas = []
+            child_codons = []
+            child_aas = []
+
+            for i in range(nsites):
+                codon_start = i * 3
+                codon_end = codon_start + 3
+
+                parent_codon = parent_seq[codon_start:codon_end]
+                parent_aa = parent_aa_seq[i] if i < len(parent_aa_seq) else "X"
```

Comment:
Can you explain this? `i` is in nucleotide land, right? But `parent_aa_seq` is not?

Reply:
Ah, of course, thank you. In that case it's not clear to me when this if will get triggered given

```
nsites == len(pcp_row["parent"]) // 3
```

```
parent_aa_seq = translate_sequence(parent_seq)
```

Is it when the sequence isn't a multiple of 3? If so then I suggest we start by asserting that the sequence is 0 mod 3 in length and then not doing the if.